api/types.proto

```diff
@@ -280,6 +280,23 @@ message ContainerStatus {
 
 	int32 pid = 2 [(gogoproto.customname) = "PID"];
 	int32 exit_code = 3;
+
+	HealthStatus health = 4;
+}
+
+// HealthStatus stores information about the container's healthcheck results.
+message HealthStatus {
+	string status = 1;
+	int32 failing_streak = 2;
```

Comment:
Let's ignore this PR for now, since we Merged basic health checks in a
different one. We'll revisit this in the next step.
On Thu, Jun 16, 2016 at 17:25 Dongluo Chen <notifications@github.com> wrote:

> In api/types.proto
> <https://github.com/docker/swarmkit/pull/975#discussion_r67446347>:
>
> > @@ -280,6 +280,23 @@ message ContainerStatus {
> >
> >  	int32 pid = 2 [(gogoproto.customname) = "PID"];
> >  	int32 exit_code = 3;
> > +
> > +	HealthStatus health = 4;
> > +}
> > +
> > +// HealthStatus stores information about the container's healthcheck results.
> > +message HealthStatus {
> > +	string status = 1;
> > +	int32 failing_streak = 2;
>
> Engine should already have it and you don't need to calculate.
>
> failing_streak actually has little value for our case. It's designed for
> Engine where no action is taken on the container upon failure. But engine
> continues counting how many times it has been failing.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/docker/swarmkit/pull/975/files/990082ae11a83897124c688ff9800eccebab1ee8#r67446347>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABySmQuGgPvj_1DtPshZ0VAMPQr5B5Vsks5qMeligaJpZM4I12K6>
> .
>